rest/routes/recentlyUsedWorkspaces.go

```diff
@@ -0,0 +1,205 @@
+		}
+	}
+
+	sendJSONResponse(w, http.StatusOK, responseBody)
+}
+
+// SaveRecentlyUsedWorkspaces grabs the recently used workspaces from the payload and stores them in the user's profile
+// in the database.
+func SaveRecentlyUsedWorkspaces(w http.ResponseWriter, r *http.Request) {
+	// Get the user's identity object.
+	user, ok := r.Context().Value(util.USER_CTX_KEY).(models.UserIdentity)
+	if !ok {
+		logrus.Errorf(`Unable to obtain the user identity from request %#v`, r)
+
+		sendJSONResponse(w, http.StatusInternalServerError, util.ErrorResponse{
+			Errors: []string{"Internal server error"},
+		})
+
+		return
+	}
+
+	// An empty body is not an acceptable request body.
+	if r.Body == http.NoBody {
+		logrus.Debug(`Returning a "bad requestWorkspaces" response to a "save recently used workspaces" requestWorkspaces because the incoming body is empty`)
+
+		sendJSONResponse(w, http.StatusBadRequest, util.ErrorResponse{
+			Errors: []string{"Request body is empty"},
+		})
+
+		return
+	}
+
+	// Attempt decoding the incoming payload.
+	var requestWorkspaces []models.Workspace
+	if err := json.NewDecoder(r.Body).Decode(&requestWorkspaces); err != nil {
+		logrus.Debugf(`unable to decode the body from the incoming "store recently used workspaces" requestWorkspaces: %s`, err)
+
+		sendJSONResponse(w, http.StatusBadRequest, util.ErrorResponse{
+			Errors: []string{"Unexpected body specified. Please send a list of workspaces."},
+		})
+
+		return
+	}
+
+	// There should be at least one workspace in the payload.
+	if len(requestWorkspaces) == 0 {
+		logrus.Debug(`Returning a "bad requestWorkspaces" response to a "save recently used workspaces" requestWorkspaces because the incoming body does not contain a single workspace we can save`)
+
+		sendJSONResponse(w, http.StatusBadRequest, util.ErrorResponse{
+			Errors: []string{"At least one workspace needs to be specified in the request"},
+		})
+
+		return
+	}
+
+	// Validate that the workspaces contain the right input.
+	for _, incomingWorkspace := range requestWorkspaces {
+		if errors := validateWorkspace(incomingWorkspace); len(errors) > 0 {
+			logrus.Debugf(`Returning a "bad requestWorkspaces" response to a "save recently used workspaces" requestWorkspaces because the input has the following validation problems: %v`, errors)
+
+			sendJSONResponse(w, http.StatusBadRequest, util.ErrorResponse{
+				Errors: errors,
+			})
+
+			return
+		}
+	}
+
+	// Try saving the most recently used workspaces in the database. Also, take into account the maximum number of
+	// recently used workspaces we want to save in the database.
+	var workspacesToSave []models.Workspace
+	if len(requestWorkspaces) > configuration.MaximumNumberRecentlyUsedWorkspaces {
+		workspacesToSave = requestWorkspaces[:configuration.MaximumNumberRecentlyUsedWorkspaces]
+	} else {
+		workspacesToSave = requestWorkspaces
+	}
+
+	if err := service.SaveRecentlyUsedWorkspaces(&user, workspacesToSave); err != nil {
```

Comment:
We don't want doplicates in the array. I don't think you are checking for that. If we already have WS with matching id, it should just be moved to the top of the array.

Reply:
Added a deduplication mechanism. The duplicated workspaces are moved to the top of the list and deduplicated.